ddmrp_report_part_flow_index/models/stock_buffer.py

```diff
@@ -8,5 +8,40 @@ class StockBuffer(models.Model):
     _inherit = "stock.buffer"
 
     flow_index_group_id = fields.Many2one(
-        "ddmrp.flow.index.group", string="Flow Index Group"
+        "ddmrp.flow.index.group", string="Flow Index Group", readonly=True
     )
+
+    def _calc_flow_index_group_id(self):
+        flow_index_reports = self.env["report.ddmrp.part.plan.flow.index"].read_group(
+            domain=[("buffer_id", "in", self.ids)],
+            fields=["order_frequency_group"],
+            groupby=["buffer_id"],
+        )
+        flow_index_groups = self.env["ddmrp.flow.index.group"].search([])
+        for rec in self:
+            flow_index_report = list(
+                filter(lambda x: x["buffer_id"][0] == rec.id, flow_index_reports)
+            )
+            frequency_group = flow_index_report[0]["order_frequency_group"]
+            for index_group in flow_index_groups:
+                if index_group.upper_range and index_group.lower_range:
+                    if (
+                        index_group.upper_range
+                        <= frequency_group
+                        <= index_group.lower_range
+                    ):
+                        rec.flow_index_group_id = index_group
+                        return
```

Comment:
This will return the method without assigning a value to all buffers, isn't it? I think you need to use `break` to exit the current for loop and continue with the following buffer ("for rec in self" loop)